### PR TITLE
Plotting deprecation warnings

### LINKED ISSUE #1359: Add warnings that plotting is unsupported
**Is your feature request related to a problem? Please describe.**
Users sometimes think that plotting is supported when it not longer is. We should add warnings to make this very clear and then eventually remove it.

### PR DESCRIPTION
**Pull request by lewisblake**
## Change Summary

Plotting has not been supported for a long time in pyaerocom and so this PR adds warnings to these functions to let users know that they are deprecated and may be removed in the future.

## Related issue number

Close #1359

## Checklist

* [x] Start with a draft-PR
* [x] The PR title is a good summary of the changes
* [x] PR is set to AeroTools and a tentative milestone
* [x] Documentation reflects the changes where applicable
* [x] Tests for the changes exist where applicable
* [x] Tests pass locally
* [x] Tests pass on CI
* [x] At least 1 reviewer is selected
* [x] Make PR ready to review

### CODE REVIEW

File: pyaerocom/plot/heatmaps.py [hunk 1]
```diff
@@ -233,6 +234,11 @@ def df_to_heatmap(
         annotation information for rows
 
     """
+    warnings.warn(
+        "This function is no longer supported. It may be removed in future versions.",
+        DeprecationWarning,
+    )
+
```
Comment (heikoklein) [inline pyaerocom/plot/heatmaps.py:241]: `_format_annot_heatmap` is a private/internal function, no warning needed.
A comment like `# matplotlib plotting deprecated` is enough.

**Review by heikoklein — CHANGES_REQUESTED** [review]
Deprecating matplotlib based plotting functions is a good idea. Both for the user, but also for us, a more descriptive warning might be more useful, e.g. `matplotlib based plotting is no longer directly supported. This class/function may be removed in future versions.`

File: pyaerocom/plot/heatmaps.py (hunk 1 above)
Comment (lewisblake) [reply pyaerocom/plot/heatmaps.py:241]: On my branch locally this is in `df_to_heatmap`. Not sure why it looks like it is in this internal function. I agree though. Hopefully github updates these changes soon...

Comment (heikoklein) [reply pyaerocom/plot/heatmaps.py:241]: Ah, I see, the github preview is wrong. When I expand the hidden code, I see he correct function.

**Review by heikoklein — APPROVED** [review]

**Comment by heikoklein** [comment]
Thanks, and approved. But not merged since it is still a draft?

**Pull request merged**